packages/gatsby/src/schema/types/sort.ts

```diff
@@ -149,7 +150,16 @@ export const getSortInput = <TSource = any, TContext = any>({
   schemaComposer: SchemaComposer<TContext>
   typeComposer: ObjectTypeComposer<TSource, TContext>
 }): InputTypeComposer<TContext> => {
-  const inputTypeComposer = typeComposer.getInputTypeComposer()
+  // toInputObjectType() will fail to convert fields of union types, e.g.
+  //   union FooBar = Foo | Bar
+  //   type Baz {
+  //     fooBar: FooBar
+  //   }
+  // Unfortunately there is no way to just exclude such fields from the input type in graphql-compose v7+,
+  // so simply replacing them with booleans as the least confusing option :/
+  const inputTypeComposer = toInputObjectType(typeComposer, {
+    fallbackType: schemaComposer.getSTC(`Boolean`),
+  })
```

Comment:
I miss https://github.com/graphql-compose/graphql-compose/blob/640a0b22cee8892ea9e347bd78f68cfbf732ba62/src/utils/toInputObjectType.js#L86-L88 :

![meme](https://user-images.githubusercontent.com/419821/107281156-99ce4d80-6a59-11eb-8e4f-f32f3a11ea38.png)

Reply:
Yes, this is confusing. But the best approach to it would be to fix `graphql-compose` itself (make it allow the old behavior). I'll work on it after higher priority work is done.